src/components/ArticleById.jsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getArticleById } from "./Api";
+
+const ArticleById = () => {
+
+  const {article_id} = useParams()
+  const [article, setArticle] = useState([])
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    getArticleById(article_id).then((response) => {
```

Comment:
you dont set isLoading to true before we make the request which could cause a delay if we were able to navigate between articles at any point